### DISCUSSION TITLE
Docs improvement thread

### DISCUSSION THREAD

**Discussion by GeorgeHulpoi** [facebook/lexical#5128] [General]
Hello!

I want to propose to add the following list in docs:

- ~A list of all methods and what are doing.~ A better way to find all functions. Personally, I didn't find what I needed until I started to click on every link.
- A guide on how to manipulate nodes in more situations: replacement, removal, grouping, etc.

If somebody thinks that something should be added to docs, you can write in this thread.

**Comment by Benbinbin** [comment]
I think we can add some minimal examples embeded inside the doc following the functions or modules to illustrate their feature or aim

I like the [documentation](https://tiptap.dev/introduction) of Tiptap which is another open source rich text editor framework based on ProseMirror

**Comment by GeorgeHulpoi** [comment]
- **insertNewAfter:** how it can be used to manipulate when you press enter inside or after a node (with example).

**Reply by JeffDemanche** [reply]
I am dealing with this at the moment; it's been confusing trying to figure out the behavior of `insertNewAfter` and `insertBefore`

**Comment by AlessioGr** [comment]
The **discrete** property (which seems to commit the editor state immediately if set to true in the update method) should also be documented.

Relevant discord discussion: https://discord.com/channels/953974421008293909/953974421486436393/1166712289353814117

**Comment by ak1394** [comment]
Looking at the quick start docs, it's not obvious that passing a function as a value of **editorState** is the best way to create editor state, if not loading existing EditorState from elsewhere.

**Comment by Benbinbin** [comment]
I believe it would be helpful to include a dedicated page that features various Lexical-related resources, such as plugins and nodes developed by the community, as well as examples of projects. This would provide users with a convenient way to explore and utilize additional tools and practical applications.

**Comment by Benbinbin** [comment]
I've noticed a time-related problem. Since the documentation is automatically generated by a tool, it's supposed to be updated at the same time as the code.

However, there's a slight delay in the versions released on npm. As a result, a small issue arises where the newly added methods in the documentation cannot actually be used in the project. This is because the npm version hasn't been updated promptly. The latest available version is v0.12.2, which was released several months ago.

In my opinion, the Lexical team should provide an experimental version to distinguish it from the stable version. Additionally, it would be helpful to clearly indicate the version number for each method in the documentation, so users know which version the method applies to.

**Comment by Benbinbin** [comment]
When we set a `ElementNode` as a `shadowRoot`, what behavior can we expect?

**Comment by Benbinbin** [comment]
Lexical has provided some built-in [COMMANDs](https://github.com/facebook/lexical/blob/main/packages/lexical/src/LexicalCommands.ts), lots of them are [linked to specific keyboard event](https://github.com/facebook/lexical/blob/main/packages/lexical/src/LexicalEvents.ts) (when key pressed by user, Lexical will dispatch some relative commands) 

We can guess which keys are relative to which commands by their name, but some COMMANDs name are not intuitive, like `MOVE_TO_START` and `MOVE_TO_END`

I think it will be better to add some explanation about their trigger key at documentation.
